pilot/model/validation_test.go

```diff
@@ -1859,3 +1928,237 @@ func TestValidateEndUserAuthenticationPolicySpecBinding(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateGateway(t *testing.T) {
+	tests := []struct {
+		name string
+		in   proto.Message
+		out  string
+	}{
+		{"empty", &routingv2.Gateway{}, "server"},
+		{"invalid message", &routingv2.Server{}, "cannot cast"},
+		{"happy",
+			&routingv2.Gateway{
+				Servers: []*routingv2.Server{{
+					Domains: []string{"foo.bar.com"},
+					Port:    &routingv2.Server_Port{Number: 7, Protocol: "http"},
+				}},
+			},
+			""},
+		{"happy multiple servers",
+			&routingv2.Gateway{
+				Servers: []*routingv2.Server{
+					{
+						Domains: []string{"foo.bar.com"},
+						Port:    &routingv2.Server_Port{Number: 7, Protocol: "http"},
+					},
+					{
+						Domains: []string{"baz.bar.com"},
+						Port:    &routingv2.Server_Port{Number: 18, Protocol: "redis"},
+					}},
+			},
+			""},
+		{"invalid server",
+			&routingv2.Gateway{
+				Servers: []*routingv2.Server{
+					{
+						Domains: []string{"foo.bar.com"},
+						Port:    &routingv2.Server_Port{Number: 7, Protocol: "http"},
+					},
+					{
+						Domains: []string{"baz.bar.com"},
+						Port:    &routingv2.Server_Port{Number: 66000, Protocol: "redis"},
+					}},
```

Comment:
We were discussing about adding support for IPs/with or without CIDRs as well right? i.e. I should be able to specify foo.bar.com, 1.1.1.1, or service short name. I see tests for foo.bar.com, but not for IPs or service short name. 

That also reminds me that for service short names, we need to be able to interpolate the service's FQDN based on the calling pod's namespace (if this is kube). I think that can happen in another PR.

Reply:
I'll update PR to add validation for IPs and allow them to be used in the field too. Name canonicalization can come in a follow on PR as it's reasonably complex by itself.